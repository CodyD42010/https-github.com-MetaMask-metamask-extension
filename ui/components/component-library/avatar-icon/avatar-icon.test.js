/* eslint-disable jest/require-top-level-describe */
import { render } from '@testing-library/react';
import React from 'react';

import { IconName } from '..';
import {
  BackgroundColor,
  IconColor,
} from '../../../helpers/constants/design-system';
import { AvatarIcon, AVATAR_ICON_SIZES } from '.';

describe('AvatarIcon', () => {
  it('should render correctly', () => {
    const { getByTestId, container } = render(
      <AvatarIcon
        iconName={IconName.SwapHorizontal}
        data-testid="avatar-icon"
      />,
    );
    expect(getByTestId('avatar-icon')).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it('should render with different size classes', () => {
    const { getByTestId } = render(
      <>
        <AvatarIcon
          iconName={IconName.SwapHorizontal}
          size={AVATAR_ICON_SIZES.XS}
          data-testid={AVATAR_ICON_SIZES.XS}
        />
        <AvatarIcon
          iconName={IconName.SwapHorizontal}
          size={AVATAR_ICON_SIZES.SM}
          data-testid={AVATAR_ICON_SIZES.SM}
        />
        <AvatarIcon
          iconName={IconName.SwapHorizontal}
          size={AVATAR_ICON_SIZES.MD}
          data-testid={AVATAR_ICON_SIZES.MD}
        />
        <AvatarIcon
          iconName={IconName.SwapHorizontal}
          size={AVATAR_ICON_SIZES.LG}
          data-testid={AVATAR_ICON_SIZES.LG}
        />
        <AvatarIcon
          iconName={IconName.SwapHorizontal}
          size={AVATAR_ICON_SIZES.XL}
          data-testid={AVATAR_ICON_SIZES.XL}
        />
      </>,
    );
    expect(getByTestId(AVATAR_ICON_SIZES.XS)).toHaveClass(
      `mm-avatar-base--size-${AVATAR_ICON_SIZES.XS}`,
    );
    expect(getByTestId(AVATAR_ICON_SIZES.SM)).toHaveClass(
      `mm-avatar-base--size-${AVATAR_ICON_SIZES.SM}`,
    );
    expect(getByTestId(AVATAR_ICON_SIZES.MD)).toHaveClass(
      `mm-avatar-base--size-${AVATAR_ICON_SIZES.MD}`,
    );
    expect(getByTestId(AVATAR_ICON_SIZES.LG)).toHaveClass(
      `mm-avatar-base--size-${AVATAR_ICON_SIZES.LG}`,
    );
    expect(getByTestId(AVATAR_ICON_SIZES.XL)).toHaveClass(
      `mm-avatar-base--size-${AVATAR_ICON_SIZES.XL}`,
    );
  });

  it('should render with added classname', () => {
    const { getByTestId } = render(
      <AvatarIcon
        iconName={IconName.SwapHorizontal}
        className="mm-avatar-icon--test"
        data-testid="classname"
      />,
    );
    expect(getByTestId('classname')).toHaveClass('mm-avatar-icon--test');
  });

  it('should render with icon', () => {
    const { getByTestId } = render(
      <AvatarIcon
        iconName={IconName.SwapHorizontal}
        iconProps={{ 'data-testid': 'avatar-icon' }}
      />,
    );

    expect(getByTestId('avatar-icon')).toBeDefined();
  });

  it('should render with success color icon and background color', () => {
    const { getByTestId } = render(
      <AvatarIcon
        iconName={IconName.SwapHorizontal}
        color={IconColor.successDefault}
        backgroundColor={BackgroundColor.successMuted}
        data-testid="success"
      />,
    );

    expect(getByTestId('success')).toHaveClass('box--color-success-default');
    expect(getByTestId('success')).toHaveClass(
      'box--background-color-success-muted',
    );
  });
  it('should forward a ref to the root html element', () => {
    const ref = React.createRef();
    render(<AvatarIcon iconName={IconName.SwapHorizontal} ref={ref} />);
    expect(ref.current).not.toBeNull();
    expect(ref.current.nodeName).toBe('DIV');
  });
});
