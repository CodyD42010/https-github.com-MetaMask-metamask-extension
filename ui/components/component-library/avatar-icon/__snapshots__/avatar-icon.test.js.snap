// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarIcon should render correctly 1`] = `
<div>
  <div
    class="box mm-text mm-avatar-base mm-avatar-base--size-md mm-avatar-icon mm-text--body-sm mm-text--text-transform-uppercase box--display-flex box--flex-direction-row box--justify-content-center box--align-items-center box--color-primary-default box--background-color-primary-muted box--rounded-full box--border-color-transparent box--border-style-solid box--border-width-1"
    data-testid="avatar-icon"
  >
    <span
      class="box mm-icon mm-icon--size-md box--display-inline-block box--flex-direction-row box--color-inherit"
      style="mask-image: url('./images/icons/swap-horizontal.svg');"
    />
  </div>
</div>
`;
